logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/UserSessionScopeProviderImpl.kt

```diff
@@ -63,7 +64,7 @@ actual class UserSessionScopeProviderImpl(
             UserDatabaseProvider(
                 appContext,
                 userIDEntity,
-                SecurityHelper(globalPreferences).userDBSecret(userId),
+                runBlocking { SecurityHelper(globalPreferences.passphraseStorage).userDBSecret(userId) },
```

Comment:
Does this solve anything?

`runBlocking` blocks the **current thread**. 

If there's two threads calling `UserSessionScopeProvider.create`, this code will still run in parallel.